### PR TITLE
Add database setup to infrastructure code

### PR DESCRIPTION
**Pull request by paulridout**
### JIRA link (if applicable) ###
N/A

### Change description ###
- Added database.tf file to set up database using terraform-module-postgresql-flexible module.
- Updated versions of Terraform and azurerm provider to enable use of terraform-module-postgresql-flexible module.
- Changed civil_vault key vault data source to reference civil-x key vaults instead of civil-shared-x key vaults.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by kiran-yenigala-hmcts — DISMISSED** [review]

File: charts/civil-sdt/values.preview.template.yaml [hunk 1]
```diff
@@ -13,7 +13,7 @@ java:
     enabled: true
   keyVaults:
     civil:
-      resourceGroup: civil
+      resourceGroup: civil-service
```
Comment (kiran-yenigala-hmcts) [inline charts/civil-sdt/values.preview.template.yaml:16]: is the changed to civil-service?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -6,5 +6,6 @@ locals {
       "Destroy Me"   = var.destroy_me
     })
   ))
+  civil_service_resource_group = "${var.product}-service-${var.env}"
   civil_shared_resource_group = "${var.product}-shared-${var.env}"
```
Comment (kiran-yenigala-hmcts) [inline infrastructure/main.tf:10]: please remove this

**Review by kiran-yenigala-hmcts — COMMENTED** [review]
Couple of comments

File: charts/civil-sdt/values.preview.template.yaml (hunk 1 above)
Comment (paulridout) [reply charts/civil-sdt/values.preview.template.yaml]: Can't find any reference to resourceGroup in [key vault documentation](https://github.com/hmcts/chart-library#keyvault-secret-csi-volumes).  Found that it's not used in other repositories (e.g. [ccd-definition-store-api](https://github.com/hmcts/ccd-definition-store-api/blob/master/charts/ccd-definition-store-api/values.preview.template.yaml#L8)), so removed it.

File: infrastructure/main.tf (hunk 2 above)
Comment (paulridout) [reply infrastructure/main.tf]: Removed both resource group local variables as now using a new civil-sdt resource group which is defined in this file.

**Pull request merged**